test/framework/src/main/java/org/elasticsearch/entitlement/bootstrap/TestEntitlementBootstrap.java

```diff
@@ -115,20 +134,31 @@ private static Collection<Path> dataDirs(Settings settings, Path homeDir) {
             : dataDirs.stream().map(TestEntitlementBootstrap::absolutePath).toList();
     }
 
+    private static Path sharedDataDir(Settings settings) {
```

Comment:
This is required for some tests, though it looks like we never grant `PATH_SHARED_DATA_SETTING` in production.
Is this a test-only thing? Or is that a bug?